### DISCUSSION TITLE
Correct way to use the embedding layer

### DISCUSSION THREAD

**Discussion by VarLad** [LuxDL/Lux.jl#883] [Q&A] [answered]
Hello, I was wondering whats the correct way to use the Embedding layer.
Looking at the [docs](https://lux.csail.mit.edu/stable/api/Lux/layers#Lux.Embedding), I tried:
```
using Lux
x = Embedding(10 => 10)
y = x([1,2,3])
z = x(5) 
```
but that doesn't work.

**Comment by avik-pal** [comment, marked as answer]
That is not the correct way to call Lux layers. See the [manual on the interface specification](https://lux.csail.mit.edu/stable/manual/interface).

```julia
x = Embedding(10 => 10)
ps, st = Lux.setup(Random.default_rng(), x)

y = x([1,2,3], ps, st)
z = x(5, ps, st)
```
